.github/workflows/benchmark.yml

```diff
@@ -0,0 +1,112 @@
+        with:
+          name: BenchmarkResults
+          path: ${{ runner.temp	}}/current/BenchmarkResults
+      - name: Export benchmark name list to job output
+        id: set-matrix
+        run: echo "::set-output name=matrix::$(jq -c '.benchmarkResults | keys' "${{ runner.temp }}/current/BenchmarkResults")"
+
+# this job will run once per benchmark and compare the current and previous values
+  compare:
+    needs: bench
+    runs-on: ubuntu-latest
+    # parallelize this job per benchmark
+    strategy:
+        matrix:
+            benchmark: ${{ fromJson(needs.bench.outputs.matrix) }}
+        # don't cancel other benchmarks when one fails
+        fail-fast: false
+  
+    steps:
+      - name: Download benchmark results from earlier in this workflow
+        uses: actions/download-artifact@v3
+        with:
+          name: BenchmarkResults
+          path: ${{ runner.temp	}}/current
+      - name: Download benchmark results from the last completed workflow
+        # TODO smarter error handling
+        continue-on-error: true
+        uses: dawidd6/action-download-artifact@v2
+        with:
+          ### FIXME
+          ### success is too narrow if we fail the job when there are regressions
+          ### completed is too broad, it will select a run no or defective artifacts
+          workflow_conclusion: completed
+          # branch main
+          # FIXME don't compare across unrelated branches
+          event: push
+          name: BenchmarkResults
+          path: ${{ runner.temp	}}/previous
+      - name: Check out benchmark comparison script
+        uses: snow-actions/sparse-checkout@v1.1.0
+        with:
+          patterns: |
+            integration/benchmark/compare_results.jq
+      - name: Compare current and previous bencmarks
+        # TODO smarter error handling
+        continue-on-error: true
+        # only run this step if all previous steps completed without error
+        if: ${{ success() }}
+        # use jq to produce a comparison between the previous and current benchmarks
+        run: >
+          jq
+          --argfile prev ${{ runner.temp }}/previous/BenchmarkResults
+          --argfile curr ${{ runner.temp }}/current/BenchmarkResults
+          --arg benchmark ${{ matrix.benchmark }}
+          -n
+          -f integration/benchmark/compare_results.jq
+          | tee "${{ runner.temp }}/BenchmarkResultsPctDelta"
+      
+      # sanitize and store the comparison results to a step output that can be read by the later steps
+      - name: Save comparison results
+        if: ${{ success() }}
+        id: results
+        run: |
+          DELTA="$(cat ${{ runner.temp }}/BenchmarkResultsPctDelta)"
+          DELTA="${DELTA//'%'/%25}"
+          DELTA="${DELTA//$'\n'/%0A}"   
+          DELTA="${DELTA//$'\r'/%0D}"
+          echo "::set-output name=delta::$DELTA"
+
+      - name: Regression Mean
+        if: always() && fromJSON(steps.results.outputs.delta).mean > 4
```

Comment:
There's a balance to be had between this and the opposite case. With comparisons to previous benchmarks we miss slowly creeping up latency. With static targets we miss big regressions if performance has improved since the last time the targets were updated. That is, if 10/s is the alert threshold now because we expect 12/s, then when we reach 20/s we would miss a regression back to 11/s if no one has updated the target from 10/s to 18/s yet.
I am unsure where to draw this line.

Reply:
We could have both eventually, but I'd like to start from having concrete target numbers. It is like having SLAs.

Reply:
We need to decide on an instance type/size before setting those targets. I don't think any currently open avenue of discussion has approached that topic yet.